consensus/polybft/stale_sequence_check.go

```diff
@@ -14,28 +14,31 @@ type staleSequenceCheck struct {
 	checkDuration      time.Duration
 	sequenceShouldStop chan struct{}
 	quit               chan struct{}
+	stopped            chan struct{}
 	getHeader          func() *types.Header
 }
 
 func newStaleSequenceCheck(logger hclog.Logger, getHeader func() *types.Header, checkDuration time.Duration) *staleSequenceCheck {
 	return &staleSequenceCheck{
-		logger:             logger,
-		currentSequence:    0,
-		mtx:                &sync.Mutex{},
-		checkDuration:      checkDuration,
-		sequenceShouldStop: make(chan struct{}, 1),
-		quit:               make(chan struct{}),
-		getHeader:          getHeader,
+		logger:          logger,
+		currentSequence: 0,
+		mtx:             &sync.Mutex{},
+		checkDuration:   checkDuration,
+		getHeader:       getHeader,
 	}
 }
 
 func (s *staleSequenceCheck) startChecking() {
+	s.stopped = make(chan struct{})
+	s.sequenceShouldStop = make(chan struct{}, 1)
+	s.quit = make(chan struct{})
 	ticker := time.NewTicker(s.checkDuration)
 	go func() {
 		for {
 			select {
 			case <-s.quit:
 				close(s.sequenceShouldStop)
+				close(s.stopped)
```

Comment:
Is `stopped` channel redundant to `quit` channel maybe? I mean I don't clearly see why we would block in the `stopChecking` until we receive the `stopped` notification (though there might be the reason, which I'm unaware atm).

Reply:
Just to make certain that all checks are finished and all channels are closed before re-using the instance.  But it is overkill when looking at it again.  I've added another test case to call quit whilst the checking is happening and its fine without it.